subsys/mgmt/osdp/src/osdp_pd.c

```diff
@@ -380,22 +466,26 @@ static int pd_decode_command(struct osdp_pd *pd, uint8_t *buf, int len)
 		if (len != CMD_KEYSET_DATA_LEN) {
 			break;
 		}
+		/* only key_type == 1 (SCBK) and key_len == 16 is supported */
```

Comment:
Does spec defines any limitation for key length  ?

Reply:
The spec defines the key size and block size to be 128 bits that give us the 16 bytes.

Reply:
In the 2.1.7 there is stated: "The Length indicates the number of bytes containing key data in the array Data[]. It is
computed as the integer value of the quantity of Key Length in bits plus 7 divided by 8. For
example, the Length shall be 16, and the Data[] array shall contain the 128-bit Secure
Channel base key (SCBK). For the 256-bit key the Length shall be 32."
Also in the 2.2 spec does not state exact 16 byte key length.

Reply:
> In the 2.1.7 there is stated: "The Length indicates the number of bytes containing key data in the array Data[]. It is
computed as the integer value of the quantity of Key Length in bits plus 7 divided by 8. For
example, the Length shall be 16, and the Data[] array shall contain the 128-bit Secure
Channel base key (SCBK). 

Yes, I am aware of this. A lot of parts of mgmt/osdp is designed with the assumption `key_size = block_size = 16` (for that matter even the spec, see last point). If that were to change in a future version (right now there isn't any), then we need to take a look all over for such assumptions and fix them. For now, this check exists to assert it.

> For the 256-bit key the Length shall be 32.

I don't have 2.1.7, I think this has been removed in the latest version 2.2 of the spec (I cannot find it in 2.15 too).

> Also in the 2.2 spec does not state exact 16 byte key length

Not directly, no. But the spec make that obvious in various places. For instance,
- the SCBK default key is defined for 128 bits only.
- the capability mask defines only a 128 bit key.
- session key, MAC, derivation process talks about 128 bit keys only.